### PR TITLE
feat(kexternallink): introduce kui tokens [KHCP-7705]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-7705

Changes:
- introduces `kui-` design tokens in `KExternalLink` component
- removes any usage of Kongponents utility classes in `KExternalLink` component template

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: src/utilities/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    // eslint-disable-next-line no-new
+    new URL(string)
+
+    // No error was thrown, so return true
+    return true
+  } catch {
+    // Invalid URL
+    return false
+  }
+}
```
Comment (adamdehaven) [inline src/utilities/helpers.ts:18]: nit: can you rename this file to `urls` or something? We currently have each utility file delegated to a single type of "thing" that it does

Comment (portikM) [reply src/utilities/helpers.ts:18]: Done

File: src/components/KExternalLink/KExternalLink.vue [hunk 2]
```diff
@@ -61,17 +47,18 @@ const iconColor = computed((): string => isMouseOver.value ? 'var(--KExternalLin
...
+    color: var(--KExternalLinkColorHover, var(--kui-color-text-primary-strong, $kui-color-text-primary-strong));
   }
 
   .kong-icon {
-    margin-left: var(--spacing-xs);
+    margin-left: var(--spacing-xs, var(--kui-space-40, $kui-space-40));
   }
 }
 </style>
+@/utilities/urls
```
Comment (adamdehaven) [inline src/components/KExternalLink/KExternalLink.vue:64]: remove this?

Comment (portikM) [reply src/components/KExternalLink/KExternalLink.vue]: Stupid VS Code import autoupdate, removed

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**